### PR TITLE
Simplify makefile

### PR DESCRIPTION
**Pull request by D4nte**

### CODE REVIEW

**Comment by D4nte** [comment]
Note to self: squash before merging.

File: Makefile [hunk 1]
```diff
@@ -1,61 +1,31 @@
...
-FMT_LOC = $(TOOLCHAIN_LOC)/bin/cargo-fmt
-TOMLFMT_LOC = $(CARGO_DIR)/bin/cargo-tomlfmt
+CARGO = $(RUSTUP) run --install $(TOOLCHAIN) cargo --color always
 
-build: build_debug
-
-## Build Environment
-# `test` is used to only run if the target does not exist
-# Instead of running if the pre-requisite is older than the target
+CLIPPY_LOC = $(shell which cargo-clippy)
```
Comment (thomaseizinger) [inline Makefile]: This would make it more consistent? :)
```suggestion
CLIPPY = $(shell which cargo-clippy)
```

File: Makefile [hunk 2]
```diff
@@ -65,22 +38,22 @@ clean:
...
+test:
 	@$(CARGO) test --all
 
-doc: build_env
+doc:
 	@$(CARGO) doc
 
-check_format: $(FMT_LOC) $(TOMLFMT_LOC)
+check_format: $(FMT) $(TOMLFMT)
 	@$(CARGO) fmt -- --check
```
Comment (thomaseizinger) [inline Makefile]: Ideally, it should be like this no?
```suggestion
	@$(FMT) -- --check
```

Comment (D4nte) [reply Makefile]: Well, not if we want to use `rustup run`... Ok, I can try better.

Comment (thomaseizinger) [reply Makefile]: It is a bit weird that we use `$(CARGO) fmt -- --check` on the existence of `cargo-fmt` in the path. We could use the `rustup run` command to also check for the presence of `rustfmt`:

image

File: Makefile [hunk 3]
```diff
@@ -1,61 +1,34 @@
...
-## Build Environment
-# `test` is used to only run if the target does not exist
-# Instead of running if the pre-requisite is older than the target
+CLIPPY = $(shell which cargo-clippy)
+FMT = $(shell which cargo-fmt)
+TOMLFMT = $(shell which cargo-tomlfmt)
 
-$(TOOLCHAIN_LOC):
-	test -e $@ || $(RUSTUP) toolchain install $(TOOLCHAIN)
+CARGO_ENV = $(HOME)/.cargo/env
```
Comment (thomaseizinger) [inline Makefile]: This doesn't seem to be used for anything :o

File: Makefile (hunk 2 above)
Comment (D4nte) [reply Makefile]: ok, I know what I want to do :)

**Comment by D4nte** [comment]
Adding everyone as a reviewer a this is a proposal for comit-network/comit-rs#1046

Let me know what you think, use it and see if we want to migrate comit-rs to a Makefile.

File: Makefile [hunk 4]
```diff
@@ -1,86 +1,50 @@
...
-	test -e $@ || $(RUSTUP) component add rustfmt --toolchain $(TOOLCHAIN)
+install_rustfmt:
+	$(RUSTUP) component list --installed --toolchain $(TOOLCHAIN) | grep -q rustfmt || $(RUSTUP) component add rustfmt --toolchain $(TOOLCHAIN)
 
-$(TOMLFMT_LOC): $(TOOLCHAIN_LOC)
-	test -e $@ || $(CARGO) install cargo-tomlfmt
-
-dev_env: build_env $(CLIPPY_LOC) $(FMT_LOC) $(TOMLFMT_LOC)
+install_tomlfmt:
+	$(CARGO) --list |grep -q tomlfmt || $(CARGO) install cargo-tomlfmt
```
Comment (luckysori) [inline Makefile]: :wrench: 
```suggestion
	$(CARGO) --list | grep -q tomlfmt || $(CARGO) install cargo-tomlfmt
```

File: Makefile [hunk 5]
```diff
@@ -1,86 +1,50 @@
...
 
 ## User install
 
-install: build_env
-	@$(CARGO) install --path .
+install:
```
Comment (luckysori) [inline Makefile]: You may already know, but this errors if the binary is already installed and asks you to pass `--force` if you want to overwrite it.

**Comment by D4nte** [comment]
> I like it a lot more than anything related to `cargo-make`, since it's so much more readable and explicit.
> 
> On the other hand, I do wonder if we need to wrap `cargo` commands. Couldn't we use [`cargo` aliases](https://doc.rust-lang.org/cargo/reference/config.html#configuration-keys) for all this?

Your link does not work for me...

**Comment by D4nte** [comment]
> > I like it a lot more than anything related to `cargo-make`, since it's so much more readable and explicit.
> > On the other hand, I do wonder if we need to wrap `cargo` commands. Couldn't we use [`cargo` aliases](https://doc.rust-lang.org/cargo/reference/config.html#configuration-keys) for all this?
> 
Hum, I don't see how you would that does it work for a `.cargo` folder inside the project directory? It does not look like it.

File: Makefile (hunk 4 above)
Comment (D4nte) [reply Makefile]: need to find a shell formatter :P

File: Makefile (hunk 5 above)
Comment (D4nte) [reply Makefile]: yes, I kind of assume a first user install but you are right, I'll add it.

**Comment by D4nte** [comment]
> Looks good to me.
> Please also update the readme.md (i.e. last line says: `Build and install: cargo install --path . create-comit-app.`

Yes, readme! I was not sure whether we do that now or leave it for later once we all used makefile a bit. Will do now.

**Comment by thomaseizinger** [comment]
Really good work @d4nte! I like it a lot :)

**Pull request merged**